src/zarr/v3/group.py

```diff
@@ -150,15 +158,25 @@ def from_dict(
     async def getitem(
         self,
         key: str,
-    ) -> Union[AsyncArray, AsyncGroup]:
+    ) -> AsyncArray | AsyncGroup:
 
         store_path = self.store_path / key
 
+        # Note:
+        # in zarr-python v2, we first check if `key` references an Array, else if `key` references
+        # a group,using standalone `contains_array` and `contains_group` functions. These functions
+        # are reusable, but for v3 they would perform redundant I/O operations.
+        # Not clear how much of that strategy we want to keep here.
```

Comment:
I don't think we want to keep it at all! 

xref: https://github.com/zarr-developers/zarr-python/pull/1739